specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keys.json

```diff
@@ -0,0 +1,636 @@
+          "description": "Not before date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "exp": {
+          "x-ms-client-name": "Expires",
+          "type": "integer",
+          "format": "int64",
+          "description": "Expiry date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "created": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Creation time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "updated": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Last updated time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "recoveryLevel": {
+          "type": "string",
+          "description": "The deletion recovery level currently in effect for the object. If it contains 'Purgeable', then the object can be permanently deleted by a privileged user; otherwise, only the system can purge the object at the end of the retention interval.",
+          "enum": [
+            "Purgeable",
+            "Recoverable+Purgeable",
+            "Recoverable",
+            "Recoverable+ProtectedSubscription"
+          ],
+          "x-ms-enum": {
+            "name": "DeletionRecoveryLevel",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "x-nullable": false
+        }
+      },
+      "description": "The object attributes managed by the Azure Key Vault service."
+    },
+    "KeyCreateParameters": {
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The tags that will be assigned to the key."
+        },
+        "properties": {
+          "$ref": "#/definitions/KeyProperties",
+          "description": "The properties of the key to be created."
+        }
+      },
+      "description": "The parameters used to create a key.",
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "Key": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/KeyProperties",
+          "description": "The properties of the key."
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "./keyvault.json#/definitions/Resource"
+        }
+      ],
+      "description": "The key resource."
+    },
+    "KeyListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Key"
+          },
+          "description": "The key resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next page of keys."
+        }
+      },
+      "description": "The page of keys."
+    },
+    "RotationPolicy": {
```

Comment:
Should this be `KeyRotationPolicy`? I'm not sure if we intend to share this type with other types of objects like secrets or certificates.

Reply:
'RotationPolicy" should be fine. "RotationPolicy" for secrets would be in secrets.json, right? and it would be in a separate resource in ARM template.